cnd/src/swap_protocols/rfc003/alice/spawner.rs

```diff
@@ -80,32 +83,93 @@ impl<T: MetadataStore, S: StateStore, C: Client> AliceSpawner
         let (sender, receiver) = mpsc::unbounded();
 
         let swap_execution = {
+            let client = Arc::clone(&self.client);
             let ledger_events = self.ledger_events.clone();
-            alice.new_state_machine(
-                ledger_events.create_ledger_events(),
-                ledger_events.create_ledger_events(),
-                self.client.clone(),
-                bob_dial_info,
-                Arc::new(sender),
-            )
+            let state_store = Arc::clone(&self.state_store);
+
+            // For legacy reasons, we also store the communication in the state store
```

Comment:
Can we have a "Will be remove with issue #xx" comment?

Reply:
I removed the comment because it is more confusing than it actually helps and I don't yet know when we are going to "fix" it. Also because I don't yet know how :)